frontend/src/hooks/secret-operations/useCreatePersonalSecret.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from "react";
+
+import { useFetchSecretValue } from "../api/dashboard/queries";
+import { SecretType } from "../api/types";
+import { useHandleSecretOperation } from "./useHandleSecretOperation";
+
+interface CreatePersonalSecretParams {
+  key: string;
+  secretPath: string;
+  environment: string;
+}
+
+export function useCreatePersonalSecretOverride(projectId: string) {
+  const handleSecretOperation = useHandleSecretOperation(projectId);
+  const fetchSecretValue = useFetchSecretValue();
+
+  const createPersonalSecretOverride = useCallback(
+    async ({ key, secretPath, environment }: CreatePersonalSecretParams, value?: string) => {
+      let overrideValue = value;
+      if (!value) {
+        // Default to using current value of the secret
+        const result = await fetchSecretValue({
+          environment,
+          projectId,
+          secretKey: key,
+          secretPath
+        });
+        overrideValue = result.value;
```

Comment:
When an override is created, by default it takes the current value of the secret.

This eliminates the need for the component creating the personal override to manage state of the value before it's created.

**Possible Bug**: Creating a personal override with an empty value also clears the original secret - the PR is already big, please verify and create an issue.